src/vanilla/ClientModelExtensions.cs

```diff
@@ -417,6 +418,15 @@ private static string AddMetaData(this IModelType type, string serializedName, I
                 defaultValue = "{}";
             }
 
+            if((bool)Settings.Instance.CustomSettings["ClientSideValidation"])

```

Comment:
it seems that this setting would be embedded in the generated code, so we either generate code that does client side validation or not - being that the case, should we directly not include in the mapper the constraints and requiredness? Do we need to have another property to indicate it? 
Are the constraints and requiredness used for something else (apart from the client side validation) in the generated code, so that we would need to have them in the mapper?

Reply:
is the default of the value from the setting coming from autorest true by default? 
If it's not indicated in the command line, the setting should be true, right?

Reply:
To answer your second question, Yes. By default the client side validation is set to true. If the command line does not have the value set, then it is set to true. 

To answer your first question, We could manipulate the mapper too. But, I do not see a good reason to do it. It is either this approach or manipulating the mapper. Both would work. I chose this for 2 reasons. 

1. This is the approach used by Node. So, I wanted consistency. 
2. The mapper object must be correct at all times irrespective of the passed values. How do you use the mapper? That logic could change. But, mapper as a whole should be consistent and correct at all times.

Reply:
1. Please sync up with Amar for this one. 
2. I do agree that the constraints are not used anywhere else. But, I do not think we should eliminate them from mapper. This might cause confusion in the future. So, I prefer to keep them as such.

Reply:
1. @amarzavery where's node picking up client side validation? @sarangan12 since you pointed out consistency, can you point to where that consistency is coming from? 
2. ok.